src/main/apiClient.ts

```diff
@@ -809,14 +816,10 @@ export default class ApiClient {
     })
   }
 
-  private getPkceParams(authParams: AuthParameters): Promise<PkceParams | {}> {
-    if (this.config.pkceEnabled) {
-      if (authParams.responseType === 'token')
-        return Promise.reject(new Error('Cannot use implicit flow when PKCE is enabled'))
-      else
-        return computePkceParams()
-    } else
-      return Promise.resolve({})
+  private static buildPkceParams(authParams: AuthParameters): Promise<PkceParams | {}> {
+    if (authParams.responseType === 'token') {
+      return Promise.reject(new Error('Cannot use implicit flow when PKCE is enforced'))
+    } else return computePkceParams()
```

Comment:
1. Nitpick on terminology: I prefer sticking to "get" as we aren't building the PKCE params using any provided arguments
2. The error states implicit cannot be used when PKCE is enforced but does not check against the configuration

In this method, we need to check that:
```javascript
if (responseType === 'code') {
  computePkceParams()
} else {
  if (authParams.responseType === 'token' && this.config.pkceEnforced) {
    return Promise.reject(new Error('Cannot use implicit flow when PKCE is enforced'))
  } else {
    return Promise.resolve({})
  }
}
```
_(after which the returned promise will still be `Promise<PkceParams | {}>` whereas it is currently just `Promise<PkceParams>`)_